zerver/views/read_receipts.py

```diff
@@ -72,4 +75,30 @@ def read_receipts(
         .values_list("user_profile_id", flat=True)
     )
 
-    return json_success(request, {"user_ids": list(user_ids)})
+    # Filter out any users with a muted relationship to the current user.
+    muted_users: Set[int] = set(
+        MutedUser.objects.filter(Q(user_profile=user_profile) | Q(muted_user=user_profile))
+        .annotate(
+            other_user=Case(
+                When(user_profile=user_profile, then="muted_user"),
+                When(muted_user=user_profile, then="user_profile"),
+            )
+        )
+        .values_list("other_user", flat=True)
+    )
```

Comment:
I think you should be able to define a reverse relation using 

https://docs.djangoproject.com/en/4.1/ref/models/fields/#django.db.models.ForeignKey.related_name

Reply:
Yeah, I read through that documentation a few times. The `MutedUser` table has the `related_name` set to "+", which disables the backwards relation.

```python
class MutedUser(models.Model):
    user_profile: UserProfile = models.ForeignKey(UserProfile, related_name="+", on_delete=CASCADE)
    muted_user: UserProfile = models.ForeignKey(UserProfile, related_name="+", on_delete=CASCADE)
    date_muted: datetime.datetime = models.DateTimeField(default=timezone_now)
```

From the linked Django documentation:

>If you’d prefer Django not to create a backwards relation, set related_name to '+' or end it with '+'. For example, this will ensure that the User model won’t have a backwards relation to this model: 
> ```
> user = models.ForeignKey(
>     User,
>     on_delete=models.CASCADE,
>     related_name='+',
> )
>```

Reply:
Right, so my suggestion is that we do a prep commit to replace `related_name='+'` with something like `related_name='muted_by_users'`. That change will require a migration, but I expect the migration to be trivial, as, if I understand correctly, it's actually a noop in the actual database -- just Django noting for itself that `UserProfile.users_muting` will secretly do the appropriate join between the tables.

I'm not sure that's the best name for the `related_name` field here; user your best judgement once you've got code actually using it for what feels readable.

Reply:
Thanks for that clarification! I've updated for the migration prep commit and now I've updated the original query to use that backward relation for excluding the users with a muted relationship to the current user. Marking as ready for review.